src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembershipInternal.java

```diff
@@ -3,24 +3,67 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.acsp.manage.users.exceptions.NotFoundRuntimeException;
+import uk.gov.companieshouse.acsp.manage.users.model.AcspMembersDao;
+import uk.gov.companieshouse.acsp.manage.users.repositories.AcspMembersRepository;
+import uk.gov.companieshouse.acsp.manage.users.service.AcspDataService;
+import uk.gov.companieshouse.acsp.manage.users.service.AcspMembersService;
+import uk.gov.companieshouse.acsp.manage.users.service.UsersService;
+import uk.gov.companieshouse.acsp.manage.users.utils.StaticPropertyUtil;
+import uk.gov.companieshouse.api.accounts.user.model.UsersList;
 import uk.gov.companieshouse.api.acsp_manage_users.api.UserAcspMembershipInternalInterface;
 import uk.gov.companieshouse.api.acsp_manage_users.model.InternalRequestBodyPatch;
 import uk.gov.companieshouse.api.acsp_manage_users.model.InternalRequestBodyPost;
 import uk.gov.companieshouse.api.acsp_manage_users.model.ResponseBodyPost;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.util.List;
+import java.util.Optional;
 
 @RestController
 public class UserAcspMembershipInternal implements UserAcspMembershipInternalInterface {
 
-    public UserAcspMembershipInternal() {}
+    private final AcspDataService acspDataService;
+    private final UsersService usersService;
+    private final AcspMembersRepository acspMembersRepository;
+    private final AcspMembersService acspMembersService;
+    private static final Logger LOG = LoggerFactory.getLogger(StaticPropertyUtil.APPLICATION_NAMESPACE);
+
+    @Autowired
+    public UserAcspMembershipInternal(AcspDataService acspDataService,
+                                      UsersService usersService,
+                                      AcspMembersRepository acspMembersRepository,
+                                      AcspMembersService acspMembersService) {
+        this.acspDataService = acspDataService;
+        this.usersService = usersService;
+        this.acspMembersRepository = acspMembersRepository;
+        this.acspMembersService = acspMembersService;
+    }
 
     @Override
     public ResponseEntity<ResponseBodyPost> addAcspOwner(
-            @NotNull String xRequestId,
-            @Pattern(regexp = "^[0-9A-Za-z-_]{0,32}$") String acspNumber,
-            @Valid InternalRequestBodyPost internalRequestBodyPost) {
-        return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1238)
+            final String xRequestId,
+            final String acspNumber,
+            final InternalRequestBodyPost internalRequestBodyPost) {
+        LOG.debugContext(xRequestId, String.format("Attempting to add ACSP owner for acsp_number %s", acspNumber), null);
+        acspDataService.fetchAcspData(acspNumber);
```

Comment:
We're not doing anything with the result of this, and it doesn't have any side effects - could we remove it?
```suggestion
```

Reply:
This call is to verify if acsp with the provided acspNumber exists. The fetchAcspData method will throw an exception if it doesn't.